### PR TITLE
Feature/adrian login

### PR DESCRIPTION
**Pull request by AdrianArcoGonzalez**

### CODE REVIEW

File: src/components/Login/Login.test.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+describe("Given a Login components", () => {
+  describe("When it's instantiated", () => {
+    test("Then it should show to inputs where the user can write", async () => {
+      const inputUserText = "Username";
+      const inputPasswordText = "Password";
+      const userWrite = "John Doe";
+      const passwordWrite = "123123";
+
+      render(
+        <BrowserRouter>
```
Comment (gerardramonp) [inline src/components/Login/Login.test.tsx:16]: You can extract this render method to a utils file, as you will use it across the whole App.

There you can config all the Providers you need and create different functions, for ex:

- reduxTestWrapperFactory(component, initialState)  --> This one will have redux provider
- formikTestWrapperFactory(component, initialFormikState, handleSumbit) --> Formik provider (so the components can use the formikContext hook

The function can be similar to this:

function whateverTestWrapperFactory(component) {
  return render(
    <BrowserRouter>
        <Provider1>
           <Provider2>
                {component}
           <Provider2>
        </Provider1>
    </BrowserRouter>
  )
}

Then you can invoke it like this:
const screen = whateverTestWrapperFactory(<MyComponent/>, {user:{logged: true}})

File: src/components/Login/Login.tsx [hunk 2]
```diff
@@ -16,37 +16,23 @@ const Login = (): JSX.Element => {
...
-            resetForm();
-          }, 2000);
+          resetForm();
         }}
       >
-        {({ errors, isSubmitting }) => (
-          <Form autoComplete="off" className="form">
+        {({ errors, isValid }) => (
+          <Form autoComplete="off" className="form" noValidate>
             <div className="input-container">
```
Comment (gerardramonp) [inline src/components/Login/Login.tsx:26]: Move all this <Form>....</Form> code into a component, and use the formikContext there.

This way you can remove the line from above 
`{({ errors, isValid }) => (`

And in the new component:
 const {errors, isValid} = useFormikContext<formSchemaType>()

File: src/components/Login/Login.tsx [hunk 3]
```diff
@@ -16,37 +16,23 @@ const Login = (): JSX.Element => {
...
               </label>
               <Field
-                type="text"
-                id="username"
                 name="username"
+                id="username"
+                type="text"
                 className="form-input"
-                placeholder="John"
+                placeholder="John Doe"
```
Comment (gerardramonp) [inline src/components/Login/Login.tsx:35]: Use a proper placeholder.

File: src/components/Login/Login.tsx [hunk 4]
```diff
@@ -16,37 +16,23 @@ const Login = (): JSX.Element => {
...
-          }, 2000);
+          resetForm();
         }}
       >
-        {({ errors, isSubmitting }) => (
-          <Form autoComplete="off" className="form">
+        {({ errors, isValid }) => (
+          <Form autoComplete="off" className="form" noValidate>
             <div className="input-container">
               <label htmlFor="username" className="form-label">
```
Comment (gerardramonp) [inline src/components/Login/Login.tsx:27]: You can create a custom component that wraps the <label> and the <Field> tabs so the form code looks cleaner and its faster to write.

```
const FormField = ({type, name, id, label, placeholder... =})=>{
   return(
       <>
          <label htmlfor={id} ...> {label} </label>
          <Field {all props}/>
       </>
   )
}
```

**Review by gerardramonp — CHANGES_REQUESTED** [review]

File: src/components/Login/Login.test.tsx (hunk 1 above)
Comment (AdrianArcoGonzalez) [reply src/components/Login/Login.test.tsx:16]: Thanks for the idea. I will take it

File: src/components/Login/Login.tsx (hunk 4 above)
Comment (AdrianArcoGonzalez) [reply src/components/Login/Login.tsx]: I just fixed it on this new pull

File: src/components/Login/Login.tsx (hunk 3 above)
Comment (AdrianArcoGonzalez) [reply src/components/Login/Login.tsx]: replaced, thank you

File: src/components/Login/Login.tsx (hunk 2 above)
Comment (AdrianArcoGonzalez) [reply src/components/Login/Login.tsx]: Just moved it all on the las pull request.

File: src/components/Login/Login.tsx [hunk 5]
```diff
@@ -1,90 +1,25 @@
...
-          return errors;
-        }}
+        initialValues={{ username, password }}
+        validationSchema={validateLoginSchema}
         onSubmit={(values, { resetForm }) => {
-          setTimeout(() => {
-            console.log(values);
-            resetForm();
-          }, 2000);
+          console.log(values);
```
Comment (ivanggv2) [inline src/components/Login/Login.tsx:18]: console.log

File: src/components/LoginFormikForm/LoginFormikForm.test.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { screen } from "@testing-library/react";
+import renderWithFormik from "../../Utils/test-utils";
+import LoginFormikForm from "./LoginFormikForm";
+
+describe("Given a LoginFormikForm componen", () => {
```
Comment (ivanggv2) [inline src/components/LoginFormikForm/LoginFormikForm.test.tsx:5]: Given a LoginFormikForm component

**Review by ivanggv2 — APPROVED** [review]

File: src/schemas/validateLoginSchema.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import * as yup from "yup";
+
+export const validateLoginSchema = yup.object().shape({
+  username: yup
+    .string()
+    .required("Name is required")
+    .min(5, "Min length is 5 characters")
```
Comment (gerardramonp) [inline src/schemas/validateLoginSchema.ts:7]: There are people with short names of only 2 characters. Or really long ones.

We should use a min of 1 and a maximum of 50 maybe.

File: src/schemas/validateLoginSchema.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    .max(12, "Max length is 12 characters"),
+
+  password: yup
+    .string()
+    .required("Password is required")
+    .min(8, "Min length is 8 characters"),
```
Comment (gerardramonp) [inline src/schemas/validateLoginSchema.ts:13]: Password must have 8 or mor characters.

File: src/components/LoginFormikForm/LoginFormikForm.test.tsx (hunk 6 above)
Comment (AdrianArcoGonzalez) [reply src/components/LoginFormikForm/LoginFormikForm.test.tsx:5]: fixed

File: src/components/Login/Login.tsx (hunk 5 above)
Comment (AdrianArcoGonzalez) [reply src/components/Login/Login.tsx:15]: Just deleted it, thank you

File: src/schemas/validateLoginSchema.ts (hunk 7 above)
Comment (AdrianArcoGonzalez) [reply src/schemas/validateLoginSchema.ts:7]: Just changed this messages for the next commit thank you

File: src/schemas/validateLoginSchema.ts (hunk 8 above)
Comment (AdrianArcoGonzalez) [reply src/schemas/validateLoginSchema.ts:13]: Fixed the message, thank you.

**Review by ivanggv2 — APPROVED** [review]

**Pull request merged**